src/cephadm/cephadm

```diff
@@ -932,9 +950,18 @@ def create_daemon_dirs(fsid, daemon_type, daemon_id, uid, gid,
             makedirs(os.path.join(data_dir_root, config_dir, 'certs'), uid, gid, 0o755)
             makedirs(os.path.join(data_dir_root, config_dir, 'provisioning/datasources'), uid, gid, 0o755)
             makedirs(os.path.join(data_dir_root, 'data'), uid, gid, 0o755)
+        elif daemon_type == 'alertmanager':
+            data_dir_root = get_data_dir(fsid, daemon_type, daemon_id)
+            config_dir = 'etc/alertmanager'
+            makedirs(os.path.join(data_dir_root, config_dir), uid, gid, 0o755)
+            makedirs(os.path.join(data_dir_root, config_dir, 'data'), uid, gid, 0o755)
+
 
         # populate the config directory for the component from the config-json
         for fname in required_config:
+            # config entries prefixed by '_' denote a setting not a config file
```

Comment:
Would it be possible to h a better, more descriptive, way to differentiate settings vs config files? As this will be part of the API of cephadm, it might be non-trivial to fix this later on.

Reply:
Sure. Any suggestions? Alertmanager is the only daemon thus far that needs to have it's command args altered instead of config - specifically for clustering. Adding another paramter just for alertmanager seems over the top, but perhaps we could using a different term in the config_json to denote args like this? What about just that - use the key of args to denote daemon runtime settings?

Reply:
maybe we can make a static list of config entries that should end up as a config is an idea?

Reply:
@p-se you opinion?

Reply:
Personally, I think it would be the clearest to extend the `config-json` entry in the `components` dictionary to contain `files` and `args`, e.g.:

```
        "alertmanager": {
            "image": {
                "image": "prom/alertmanager",
                "cpus": "2",
                "memory": "2GB",
                "args": [],
            },
            "config-json": { 
                "files": [
                    "alertmanager.yml",
                ],
                "args": [
                    "peers",
                ] 
            } 
        },
```

This would require us to adapt the configuration files to look different:

```
{           
    "files": {
        "alertmanager.yml": [
            "global:",
            "    resolve_timeout: 5m",
            "route:",
            "    group_by: ['alertname']",
            "    group_wait: 10s",
            "    group_interval: 10s",
            "    repeat_interval: 1h",
            "    receiver: 'ceph-dashboard'",
            "receivers:",
            "- name: 'ceph-dashboard'",
            "  webhook_configs:",
            "  - url: https://bootstrap.storage.lab:8443/api/prometheus_receiver"
        ]
    },
    "args": {
        "peers":[
            "10.90.90.150:9094",
            "10.90.90.146:9094"
        ]
    }
}
```

I also don't see the necessity to have it implemented for just once case, but considering the possible difficulty in changing it aftwards, I'd prefer to have it implemented.